java/kotlin-extractor2/src/main/kotlin/entities/TypeArgument.kt

```diff
@@ -0,0 +1,69 @@
+package com.github.codeql.entities
+
+import com.github.codeql.*
+import org.jetbrains.kotlin.analysis.api.KaSession
+import org.jetbrains.kotlin.analysis.api.types.KaStarTypeProjection
+import org.jetbrains.kotlin.analysis.api.types.KaTypeArgumentWithVariance
+import org.jetbrains.kotlin.analysis.api.types.KaTypeProjection
+import org.jetbrains.kotlin.types.Variance
+
+context(KaSession)
+fun KotlinUsesExtractor.getTypeArgumentLabel(arg: KaTypeProjection): TypeResultWithoutSignature<DbReftype> {
+
+    fun extractBoundedWildcard(
+        wildcardKind: Int,
+        wildcardLabelStr: String,
+        wildcardShortName: String,
+        boundLabel: Label<out DbReftype>
+    ): Label<DbWildcard> =
+        tw.getLabelFor(wildcardLabelStr) { wildcardLabel ->
+            tw.writeWildcards(wildcardLabel, wildcardShortName, wildcardKind)
+            tw.writeHasLocation(wildcardLabel, tw.unknownLocation)
+            tw.getLabelFor<DbTypebound>("@\"bound;0;{$wildcardLabel}\"") {
+                tw.writeTypeBounds(it, boundLabel, 0, wildcardLabel)
+            }
+        }
+
+    // Note this function doesn't return a signature because type arguments are never
+    // incorporated into function signatures.
+    return when (arg) {
+        is KaStarTypeProjection -> {
```

Comment:
This is all basically unchanged from the IR format -- the representation of variances is essentially the same. Basic translation to wildcards (`out T` == `? extends T`, `in T` == `? super T`) was likewise restored.